test/modules/SessionReplay.spec.ts

```diff
@@ -36,4 +36,15 @@ describe('Session Replay Module', () => {
     expect(NativeSessionReplay.getSessionReplayLink).toBeCalledTimes(1);
     expect(NativeSessionReplay.getSessionReplayLink).toReturnWith('link');
   });
+
+  it('should call the native method setSyncCallback', () => {
+    const callback = jest.fn().mockReturnValue(true);
+
+    SessionReplay.setSyncCallback(callback);
+    emitter.emit(NativeEvents.SESSION_REPLAY_ON_SYNC_CALLBACK_INVOCATION);
+
+    expect(NativeSessionReplay.setSyncCallback).toBeCalledTimes(1);
+    expect(emitter.listenerCount(NativeEvents.SESSION_REPLAY_ON_SYNC_CALLBACK_INVOCATION)).toBe(1);
+    expect(NativeSessionReplay.evaluateSync).toBeCalledTimes(1);
```

Comment:
Let's assert that `evaluateSync` gets called with the correct value as well.

```suggestion
    const shouldSync = true;
    const callback = jest.fn().mockReturnValue(shouldSync);

    SessionReplay.setSyncCallback(callback);
    emitter.emit(NativeEvents.SESSION_REPLAY_ON_SYNC_CALLBACK_INVOCATION);

    expect(NativeSessionReplay.setSyncCallback).toBeCalledTimes(1);
    expect(emitter.listenerCount(NativeEvents.SESSION_REPLAY_ON_SYNC_CALLBACK_INVOCATION)).toBe(1);
    expect(NativeSessionReplay.evaluateSync).toBeCalledTimes(1);
    expect(NativeSessionReplay.evaluateSync).toBeCalledWith(shouldSync);
```